zilencer/management/commands/rundjangoserver.py

```diff
@@ -1,69 +1,46 @@
-import errno
-import os
-import socket
-import sys
-from typing import Any, NoReturn
+from datetime import datetime
 
-from django.conf import settings
 from django.core.management.commands.runserver import Command as DjangoCommand
-from django.core.servers.basehttp import run
-from django.utils import autoreload
+from typing_extensions import Protocol
 
 
-class Command(DjangoCommand):
-    # Copied from Django's runserver and modified to remove print statements.
-    # This lets us declutter run-dev.py startup output.
-    def inner_run(self, *args: Any, **options: Any) -> None:
-        # If an exception was silenced in ManagementUtility.execute in order
-        # to be raised in the child process, raise it now.
-        autoreload.raise_last_exception()
+class Writable(Protocol):
+    def write(self, s: str) -> None:
+        ...
 
-        threading = options["use_threading"]
-        # 'shutdown_message' is a stealth option.
-        shutdown_message = options.get("shutdown_message", "")
-        quit_command = "CTRL-BREAK" if sys.platform == "win32" else "CONTROL-C"
 
-        self.check(display_num_errors=False)
-        # Need to check migrations here, so can't use the
-        # requires_migrations_check attribute.
-        self.check_migrations()
-        self.stdout.write(
-            ("Django process (re)started. Quit the server with %(quit_command)s.\n")
-            % {
-                "version": self.get_version(),
-                "settings": settings.SETTINGS_MODULE,
-                "protocol": self.protocol,
-                "addr": "[%s]" % self.addr if self._raw_ipv6 else self.addr,
-                "port": self.port,
-                "quit_command": quit_command,
-            }
-        )
+class FakeStdout:
+    """
+    Filter stdout from Django's runserver, to declutter run-dev.py
+    startup output.
+    """
 
-        try:
-            handler = self.get_handler(*args, **options)
-            run(
-                self.addr,
-                int(self.port),
-                handler,
-                ipv6=self.use_ipv6,
-                threading=threading,
-                server_cls=self.server_cls,
+    def __init__(self, stdout: Writable) -> None:
+        self.stdout = stdout
+        # Might fail to suppress the date line around midnight, but, whatever.
+        self.date_prefix = datetime.now().strftime("%B %d, %Y - ")
```

Comment:
Can you add an example of the output this line is supposed to catch in a comment?  

(I'd like it to be obvious whether anything not printed by this startup logic could be suppressed by this logic)

Reply:
Without filtering, the output is

```
Performing system checks...

System check identified no issues (2 silenced).
May 25, 2021 - 01:10:48
Django version 3.2.2, using settings 'zproject.settings'
Starting development server at http://127.0.0.1:9992/
Quit the server with CONTROL-C.
```

(Here’s the [source](https://github.com/django/django/blob/stable/3.2.x/django/core/management/commands/runserver.py).)

Reply:
OK, makes sense.